### PR TITLE
core: Reserve CXI provider constants to avoid conflicts

### PR DESCRIPTION
**Pull request by swelch**
Reserve address and protocol constants for the CXI provider
to avoid conflicts in applications that may be using a
pre-upstream release of the provider with other provider
definitions.

Signed-off-by: Steve Welch <welch@hpe.com>

### CODE REVIEW

**Review by shefty — APPROVED** [review]
Looks reasonable to me.  Waiting for CI

**Comment by swelch** [comment]
> Looks reasonable to me. Waiting for CI

Once  CI completes/PR is merged, I will cherry pick into v1.15.x if you are ok with that.

**Comment by wzamazon** [comment]
Usually we do not back port new features.

**Comment by swelch** [comment]
> Usually we do not back port new features.

Yes I am aware and why I asked. However, this is a simple reservation that will allow libfabric applications built using v1.15.x to make compile time decisions; it does not add features in itself to the libfabric library.

**Comment by shefty** [comment]
Backporting the change makes it easier for an out of tree provider to work with a stable release or branch.  And the change itself doesn't really bother me.  But the v1.16 release is only a couple months away, then that will become the stable branch/release.  Is there a justification for why you want this added to v1.15.x, versus waiting for v1.16?  Is cxi being developed based on v1.15?  I don't understand what compile time check this would help with, since the provider isn't necessarily present.

I guess this becomes a question for how older releases should/can support new providers.  Obviously, we wouldn't merge the provider itself, but it can install/work as a DL provider.  Are we okay accepting the new enum values, which are pretty harmless?  I don't have a strong opinion against this myself.

**Pull request merged**
